phoenix-core/src/it/java/org/apache/phoenix/end2end/transform/TransformToolIT.java

```diff
@@ -948,6 +959,175 @@ public void testTransformVerify_ForceCutover() throws Exception {
+            stmt1.setInt(3, 95051);
+            stmt1.setString(4, "");
+            stmt1.setInt(5, 101);
+            stmt1.setString(6, "viewCol12");
+            stmt1.executeUpdate();
+            tenantConn1.commit();
+        }
+
+        try (Connection tenantConn2 = getTenantConnection("tenant2")) {
+            PreparedStatement stmt1 = tenantConn2.prepareStatement(String.format(upsertQuery, view2Name));
+            stmt1.setInt(1, 11);
+            stmt1.setString(2, "uname11");
+            stmt1.setInt(3, 950511);
+            stmt1.setString(4, "");
+            stmt1.setInt(5, 111);
+            stmt1.setString(6, "viewCol22");
+            stmt1.executeUpdate();
+            tenantConn2.commit();
+        }
+
+        try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+            conn.createStatement().execute("ALTER TABLE " + dataTableFullName +
+                    " SET IMMUTABLE_STORAGE_SCHEME=SINGLE_CELL_ARRAY_WITH_OFFSETS, COLUMN_ENCODED_BYTES=2");
+            SystemTransformRecord record = Transform.getTransformRecord(schemaName, dataTableName, null, null, conn.unwrap(PhoenixConnection.class));
+            assertNotNull(record);
+            assertMetadata(conn, PTable.ImmutableStorageScheme.SINGLE_CELL_ARRAY_WITH_OFFSETS, PTable.QualifierEncodingScheme.TWO_BYTE_QUALIFIERS, record.getNewPhysicalTableName());
+
+            List<String> args = getArgList(schemaName, dataTableName, null,
+                    null, null, null, false, false, false, false, false);
+            runTransformTool(args.toArray(new String[0]), 0);
+            Transform.doCutover(conn.unwrap(PhoenixConnection.class), record);
+            Transform.updateTransformRecord(conn.unwrap(PhoenixConnection.class), record, PTable.TransformStatus.COMPLETED);
+            try (Admin admin = conn.unwrap(PhoenixConnection.class).getQueryServices().getAdmin()) {
+                admin.disableTable(TableName.valueOf(dataTableFullName));
+                admin.truncateTable(TableName.valueOf(dataTableFullName), true);
+            }
+        }
+
+        try (Connection tenantConn1 = getTenantConnection("tenant1")) {
+            String sql = "SELECT VIEW_COL11, VIEW_COL12 FROM %s ";
+            ResultSet rs1 = tenantConn1.createStatement().executeQuery(String.format(sql, view1Name));
+            assertTrue(rs1.next());
+            assertEquals(101, rs1.getInt(1));
+            assertEquals("viewCol12", rs1.getString(2));
+        }
+
+        try (Connection tenantConn2 = getTenantConnection("tenant2")) {
+            String sql = "SELECT VIEW_COL21, VIEW_COL22 FROM %s ";
+            ResultSet rs1 = tenantConn2.createStatement().executeQuery(String.format(sql, view2Name));
+            assertTrue(rs1.next());
+            assertEquals(111, rs1.getInt(1));
+            assertEquals("viewCol22", rs1.getString(2));
+        }
+    }
+
+
+    public static Connection getTenantConnection(String tenant) throws SQLException {
```

Comment:
Could we please get a test for overlapping but still updatable views? (For example, one view is WHERE Foo = 'a' AND Bar = 'b', and another is just WHERE Foo = 'a'.)

Reply:
In TransformMonitorExtendedIT I believe you have an updatable and non-updatable tenant view that overlap, but since we skip transforming read-only views, that's not quite the same thing.

Reply:
Sure. Will add one.